pkg/wal/processor/transformer/wal_transformer_validator.go

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package transformer
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/jackc/pgx/v5/pgtype"
+	pglib "github.com/xataio/pgstream/internal/postgres"
+	"github.com/xataio/pgstream/pkg/transformers"
+	"golang.org/x/exp/slices"
+)
+
+func GetTransformerValidationFunc(ctx context.Context, pgurl, snapshoturl string) func(*Transformer) error {
```

Comment:
This should be done in the stream run, where we decide if we pass on the validation option to the transformer or not. You'd expose the `pgTransformerValidator` function and use it directly there when needed.

You have to think that the wal processor transformer layer doesn't know anything about pg urls or snapshot urls and they don't really mean anything in this context (it's leaking configuration details into the package).